Lib/test/test_pathlib.py

```diff
@@ -2478,6 +2478,220 @@ def test_complex_symlinks_relative(self):
     def test_complex_symlinks_relative_dot_dot(self):
         self._check_complex_symlinks(os.path.join('dirA', '..'))
 
+class WalkTests(unittest.TestCase):
+    """Tests for Path.walk()
+
+    They are mostly just copies of os.walk tests converted to use Paths
+    because os.walk tests are already mature and cover many edge cases
+    that we could miss writing new tests.
+
+    It is not combined with Path* tests because its setup is aimed at
+    testing a bit more complex cases than Path's setup.
+    """
+
+    def setUp(self):
+        P = pathlib.Path
```

Comment:
If you're going to take a naming shortcut, `P` is not descriptive enough. You could do `cls = pathlib.Path` or `Path = pathlib.Path` at the class level and use `self.cls` or `self.Path`, respectively. Otherwise you could do `Path = pathlib.Path`.

Reply:
Great idea! Will do.